libraries/adaptive-expressions/tests/expressionParser.test.js

```diff
@@ -837,35 +833,49 @@ const scope = {
 
 describe('expression parser functional test', () => {
     it('should get right evaluate result', () => {
-        for (const data of dataSource) {
-            const input = data[0].toString();
-            console.log(input);
-            var parsed = Expression.parse(input);
-            assert(parsed !== undefined);
-            var {value: actual, error} = parsed.tryEvaluate(scope);
-            assert(error === undefined, `input: ${input}, Has error: ${error}`);
-
-            let expected = data[1];
-
-            // There's an issue when running this test near the end of an hour,
-            // where expected 'HH' doesn't always match up. This ensures it's synced with the current hour.
-            if (input === 'utcNow(\'MM-DD-YY HH\')' && expected === 'getNowTime') {
-                expected = moment(new Date().toISOString()).utc().format('MM-DD-YY HH');
-            }
-
-            assertObjectEquals(actual, expected);
-
-            //Assert ExpectedRefs
-            if (data.length === 3) {
-                const actualRefs = parsed.references();
-                assertObjectEquals(actualRefs.sort(), data[2].sort());
+        function iterateAndCheckData(source) {
+            for (const data of source) {
+                const input = data[0].toString();
+                console.log(input);
+                var parsed = Expression.parse(input);
+                assert(parsed !== undefined);
+                var {value: actual, error} = parsed.tryEvaluate(scope);
+                assert(error === undefined, `input: ${input}, Has error: ${error}`);
+    
+                let expected = data[1];
+    
+                assertObjectEquals(actual, expected);
+    
+                //Assert ExpectedRefs
+                if (data.length === 3) {
+                    const actualRefs = parsed.references();
+                    assertObjectEquals(actualRefs.sort(), data[2].sort());
+                }
+    
+                //ToString re-parse
+                const newExpr = Expression.parse(parsed.toString());
+                const newActual = newExpr.tryEvaluate(scope).value;
+                assertObjectEquals(actual, newActual);
             }
-
-            //ToString re-parse
-            const newExpr = Expression.parse(parsed.toString());
-            const newActual = newExpr.tryEvaluate(scope).value;
-            assertObjectEquals(actual, newActual);
         }
+
+        // Any tests related to getting the current time should be instantiated in the actual test.
+        // If `new Date()` is called near the end of a time change (end of an hour or end of a day),
+        // this can intermittently result in failures because the test may run at the start of an hour or day.
+        iterateAndCheckData(dataSource);
+        const timeSensitiveData = [
+            ['utcNow(\'MM-DD-YY HH\')', moment(new Date().toISOString()).utc().format('MM-DD-YY HH')],
```

Comment:
This change should be _mostly_ okay, as this expression is evaluated immediately after the creation...